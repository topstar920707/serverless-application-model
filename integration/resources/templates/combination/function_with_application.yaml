Conditions:
  TrueCondition:
    Fn::Equals:
      - true
      - true
  FalseCondition:
    Fn::Equals:
      - true
      - false

Resources:
  MyLambdaFunctionWithApplication:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs12.x
      CodeUri: ${codeuri}
      Environment:
        Variables:
          TABLE_NAME:
            Fn::GetAtt: ["MyNestedApp", "Outputs.TableName"]

  MyNestedApp:
    Type: AWS::Serverless::Application
    Condition: TrueCondition
    Properties:
      Location: ${templateurl}

  MyNestedAppFalseCondition:
    Type: AWS::Serverless::Application
    Condition: FalseCondition
    Properties:
      Location: ${templateurl}