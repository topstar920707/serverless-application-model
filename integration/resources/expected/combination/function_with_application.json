[
  { "LogicalResourceId":"MyNestedApp", "ResourceType":"AWS::CloudFormation::Stack" },
  { "LogicalResourceId":"MyLambdaFunctionWithApplication", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionWithApplicationRole", "ResourceType":"AWS::IAM::Role" }
]